core/domain/user_services.py

```diff
@@ -2821,3 +2822,88 @@ def log_username_change(committer_id, old_username, new_username):
     audit_models.UsernameChangeAuditModel(
         id=model_id, committer_id=committer_id, old_username=old_username,
         new_username=new_username).put()
+
+
+def get_user_id_from_subject_id(subject_id):
+    """Returns the user ID associated with the given subject ID.
+
+    Args:
+        subject_id: str. The subject ID.
+
+    Returns:
+        str|None. The user ID associated with the given subject ID, or None if
+        no association exists.
+    """
+    model = auth_services.UserIdBySubjectIdModel.get_by_id(subject_id)
```

Comment:
We are trying to get rid of using `get_by_id` in favor of `get`.
```suggestion
    model = auth_services.UserIdBySubjectIdModel.get(subject_id, strict=False)
```

Reply:
Done.